### ISSUE TITLE
Source map can not be fully loaded when there is only one test case

### ISSUE THREAD

**Issue by hsuehic** [bgotink/playwright-coverage#13]
I found an interested issue.  It works fine when there are more than 1 test cases, however, it can not work if there is only one test case. 

2 test cases, even if there is one empty test case, the result case be as expected:
<img width="1390" alt="image" src="https://user-images.githubusercontent.com/7324648/197509710-381902f6-55fb-4ebb-858a-708e3220ba9e.png">

1 test case, will encounter the following issue. 
<img width="1372" alt="image" src="https://user-images.githubusercontent.com/7324648/197509867-5da0ca02-98a4-4b6f-be27-acc166024926.png">

Could you kindly help to take a look?.
For reproducing:  https://github.com/hsuehic/playwright-react-typescript
`master` branch for 2-test-cases scenario.
`issue/cannot-with-one-test-case` branch for 1-test-case scenario

**Comment by bgotink** [comment]
Even with two tests it fails if the second test exits too quickly. If you remove the `test.beforeEach` call and move `await page.goto("/")` to the start of the one test with content, you'll get

```
  2 passed (4s)
Error in reporter Error [FetchError]: Invalid response body while trying to fetch http://localhost:3000/static/js/bundle.js.map: Premature close
```

I believe the cause is the timing of when the source maps are fetched. This package starts fetching sourcemaps in the `onTestEnd` reporter callback that playwright calls whenever a test ends. That callback is not awaited by playwright, so playwright continues and starts with the next test or it starts with the teardown if it was the last test.

Playwright does await the result of the final `onEnd` callback in the reporter, so this package can do its asynchronous work before playwright shuts down. However, it appears that by that time playwright has already stopped the server that it started via the `webServer.command` option.
In short: we have a fun little race condition here, and not one that can be solved by this package.

This package only downloads the source maps etc. once so if there are more tests, the initial tests will trigger a download of relevant files and subsequent tests reuse those downloads. Having more tests therefore "solves" the issue by changing the timings of the race condition so it isn't actually a race anymore.

If you're running into this issue, a possible solution is to change how the server is started. For example, the [`@ngx-playwright/test`](https://github.com/bgotink/ngx-playwright) package only starts playwright once the Angular application server is ready.

**Comment by jdell64** [comment]
I think I had the same issue. Sourcemaps came back with keys and no values.

I am making a template project that new projects can build from. I only had one example test which just goes to "/" but that completes two quickly. I added three more just like it but that completed too quickly, so I added the line

`await new Promise(resolve => setTimeout(resolve, 2000));`

to one line and it works. 

I realize this won't be an issue when I have real tests, but would certainly be a nice fix. Is there no way to check to see if the sourceMaps have been resolved and just wait?

**Comment by hsuehic** [comment]
Thanks for the detailed explanation. Close the issue, since it is beyond the scope of the package.
